internal/expr/prepare.go

```diff
@@ -108,28 +108,35 @@ type retBuilder struct {
 }
 
 // prepareExpr checks that an input or output part is correctly formatted, that
-// it corrosponds to known types and then generates the columns to go in the query.
+// it corresponds to known types and then generates the columns to go in the query.
 func prepareExpr(ti typeNameToInfo, p *ioPart) ([]fullName, []loc, error) {
 
-	var info *info
-	var ok bool
-
 	// res stores the list of columns to put in the query and their locations.
 	res := retBuilder{}
 
 	// add prepares a location and column.
 	add := func(typeName string, tag string, col fullName) error {
-		info, ok = ti[typeName]
+		in, ok := ti[typeName]
+
+		// Could not find a registered type info
 		if !ok {
+			// Handle the M type
```

Comment:
Nice! Added a variable declaration at the top of `prepareExpr()`. This will save us from doing `reflect.TypeOf(M{}).Name()` repetitively. Couldn't do an `if`, as the compared value on the left varies.